File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b-mlx_think/20260910_073214/python/002_word_frequency/solution.py
import re
from collections import Counter

with open('input/text.txt', 'r') as f:
    text = f.read()

# Extract only sequences of letters (strips all punctuation)
words = re.findall(r'[a-zA-Z]+', text)

# Convert to lowercase
words = [w.lower() for w in words]

# Count frequencies
counts = Counter(words)

# Sort by count descending, then alphabetically ascending for ties
sorted_words = sorted(counts.items(), key=lambda x: (-x[1], x[0]))

# Output
for word, count in sorted_words:
    print(f"{word}: {count}")

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b-mlx_think/20260910_073214/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const inputPath = path.resolve('input/text.txt');
const content = fs.readFileSync(inputPath, 'utf-8');

// Split into tokens by whitespace, strip non-letters, lowercase
const words = content
  .split(/\s+/)
  .map((word) => word.replace(/[^a-z]/gi, '').toLowerCase())
  .filter((word) => word.length > 0);

// Count frequency
const freq = new Map<string, number>();
for (const word of words) {
  freq.set(word, (freq.get(word) || 0) + 1);
}

// Sort by count descending, then alphabetically ascending
const sorted = Array.from(freq.entries()).sort((a, b) => {
  if (b[1] !== a[1]) return b[1] - a[1];
  return a[0].localeCompare(b[0]);
});

// Output
for (const [word, count] of sorted) {
  console.log(`${word}: ${count}`);
}